### DISCUSSION TITLE
Framework Master Category Fields

### DISCUSSION THREAD

**Discussion by vinod303** [Sunbird-Knowlg/Community#135] [General Discussions]
Team,
Ref. https://knowlg.sunbird.org/learn/product-and-developer-guide/taxonomy-and-tagging/framework-service/faqs#how-to-create-a-framework
In the above page, in the section "Apis to Create Master Categories",  there are 4 mandatory fields
 **orgIdFieldName** 
 **targetIdFieldName** 
**searchIdFieldName** 
 **searchLabelFieldName** 
 
 Would like to understand what is the significance/importance of these field?
 Will this be used in / or by any other services? 
 Since it is mandatory, i am not able to create any custom categories. Hence the question.
 @vinukumar-vs @Krishnaj20

**Comment by pallakartheekreddy** [comment]
@vinod303 We are using these fields to enable the enhanced search capability. When an object (Content, Question, Collection, Question Set) created/updated with framework categories information, the platform generates and populates these fields as additional metadata for the object.

**Reply by vinod303** [reply]
Thanks @pallakartheekreddy 
So it is a mandatory field?

**Reply by pallakartheekreddy** [reply]
Yes @vinod303
